src/expr.rs

```diff
@@ -455,9 +454,19 @@ fn rewrite_closure(capture: ast::CaptureBy,
 
     // We couldn't format the closure body as a single line expression; fall
     // back to block formatting.
-    let body_rewrite = inner_block.rewrite(&context, budget, Indent::empty());
+    let body_rewrite = try_opt!(inner_block.rewrite(&context, budget, Indent::empty()));
 
-    Some(format!("{} {}", prefix, try_opt!(body_rewrite)))
+    let block_threshold = context.config.closure_block_indent_threshold;
```

Comment:
That doesn't really work. Block formatting will always be more compact because it means more horizontal space. The heuristic is making the subjective decision between a small closure where it's easier to be inline with the current indentation (i.e., it's not worth the effort for the eye to move left) vs a big closure where the large amount of wasted whitespace becomes offensive (and it's worth moving the eye left because there's enough of a closure to justify it).